Blender-Shader-Exporter/Blender-Shader-Exporter.py

```diff
@@ -0,0 +1,472 @@
+import bpy
+import json
+
+from bpy.types import GizmoGroupProperties, IntProperty, Operator
+
+class Ops():
+    """A class to host Static Methods"""
+    @staticmethod
+    def json2File(filename: str, jsonString: str):
+
+        f = open(f"{filename}.json", "w")
+        f.write(jsonString)
+        pass
+
+    @staticmethod
+    def serialize2Json(serializableField):
+        return json.dumps(serializableField)
+
+    @staticmethod
+    def print_IgonredMaterial(materialName: str):
+        print(f"[x] Ignored  material: {materialName}")
+
+class object3D():
+    name = ""
+
+class Texture():
+    texture: str
+    isAlpha: bool
+
+    def __init__(self):
+        self.texture = ""
+        self.isAlpha = False
+
+
+class TextureSet():
```

Comment:
You didn't define specular, if this item is not defined and its functions and specifications are not applied, the correct output will not be created

Specular maps are used to define the shininess/highlight on a surface. Typically a specular map is a black and white image that maps out the shininess value on an object. The whiter the pixel, the more shiny the object in that specific place on the texture (or on the texel).